### PR TITLE
[logits] Fuse LM-head all-gather permute-copy into next_token_logits_buffer

### PR DESCRIPTION
**Pull request by cctry**
## Problem

All-gather for vocab-parallel creates two `[num_tokens, vocab]` intermediate tensors:
```
output_tensor = torch.empty(
    output_size, dtype=input_.dtype, device=input_.device
)  # new [tp * num_tokens, vocab/tp]

output_tensor = output_tensor.reshape((world_size,) + input_size)
output_tensor = output_tensor.movedim(0, dim)  # non-contiguous
output_tensor = output_tensor.reshape(
    input_size[:dim] + (world_size * input_size[dim],) + input_size[dim + 1 :]
)  # new [num_tokens, tp, vocab/tp]
```

## Fix

Let NCCL write to a pre-allocated, shared buffer, then transpose-copy into the existing `next_token_logits_buffer`. Falls back to the generic all-gather when the buffer is absent or the layout doesn't match.

## Result

Rank 0, standard CUDA-graph runner, same 52-bucket capture config (tp4, fp32 lm head):

| Metric | Before | After |
|---|---:|---:|
| Graph-pool peak live | 937.3 MB | 744.7 MB (−21%) |
| Graph-pool reserved | 1,619 MB | 789 MB (−51%) |

## Before

_(figure to be added)_

## After

_(figure to be added)_

---
### CI States

Latest PR Test (Base): :no_entry_sign: [Run #27386945411](https://github.com/sgl-project/sglang/actions/runs/27386945411)
Latest PR Test (Extra): :x: [Run #27386945298](https://github.com/sgl-project/sglang/actions/runs/27386945298)

### CODE REVIEW

File: python/sglang/srt/layers/logits_processor.py [hunk 1]
```diff
@@ -260,6 +264,28 @@ def compute_dp_attention_metadata(self):
...
 
+# Workspace for _all_gather_logits, shared by all CUDA graphs.
+_gather_logits_workspace: Optional[torch.Tensor] = None
+
+
+def get_gather_logits_workspace(
+    shape: Tuple[int, ...], dtype: torch.dtype, device: torch.device
+) -> Optional[torch.Tensor]:
+    """The workspace viewed as ``shape``/``dtype``, or None if it doesn't fit."""
+    global _gather_logits_workspace
```
Comment (ch-wan) [inline python/sglang/srt/layers/logits_processor.py:275]: This is a large persistent buffer. Would it increase peak memory when MoE is the memory bottleneck?

Comment (cctry) [reply python/sglang/srt/layers/logits_processor.py:275]: No, previously we create those tensors from graph memory pool and baked into the graph, so it is already persistent.

**Comment by merrymercy** [comment]
will think about a better solution later
